src/mcp/server/fastmcp/tools/tool_manager.py

```
@@ -68,6 +68,13 @@ def add_tool(
         self._tools[tool.name] = tool
         return tool
 
+    def remove_tool(self, name: str) -> None:
+        """Remove a tool by name."""
```

We should probably say a word about the beahviour if the tool doesn't exists.

---

I think this should raise if not found.